pkg/controller/worker/helper.go

```diff
@@ -34,8 +34,12 @@ func (w *workerDelegate) decodeWorkerProviderStatus() (*api.WorkerStatus, error)
 		return workerStatus, nil
 	}
 
-	if _, _, err := w.Decoder().Decode(w.worker.Status.ProviderStatus.Raw, nil, workerStatus); err != nil {
-		return nil, errors.Wrapf(err, "could not decode WorkerStatus '%s'", kutil.ObjectName(w.worker))
+	marshalled, err := w.worker.Status.GetProviderStatus().MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	if _, _, err := w.Decoder().Decode(marshalled, nil, workerStatus); err != nil {
+		return nil, errors.Wrapf(err, "could not decode WorkerStatus %q", kutil.ObjectName(w.worker))
```

Comment:
If you mean why using the `	marshalled, err := w.worker.Status.GetProviderStatus().MarshalJSON()` the straight answer is that it makes testing a bit easier. 

The `MarshalJson` function unifies RawExtensions's `Raw` and `Object` by returning the `Raw` is not empty and otherwise marshals the `Object` and returns that. That way I can skip the few lines of code in tests needed to write a struct/instance and marshal it into Raw. 

Normally on production only the Raw is populated, so the behaviour is the same. For all the other places tests or internal use, I think its great that I do not need to check where the data are (raw vs object) and always make sure I can access whatever is in a RawExtension without extra checks

Reply:
Interesting, thank you!